swaybar/bar.c

```diff
@@ -31,16 +32,30 @@ static void bar_init(struct bar *bar) {
 
 static void spawn_status_cmd_proc(struct bar *bar) {
 	if (bar->config->status_command) {
-		int pipefd[2];
-		if (pipe(pipefd) != 0) {
-			sway_log(L_ERROR, "Unable to create pipe for status_command fork");
+		int pipe_read_fd[2];
+		int pipe_write_fd[2];
+
+		if (pipe(pipe_read_fd) != 0) {
+			sway_log(L_ERROR, "Unable to create pipes for status_command fork");
+			return;
+		}
+		if (pipe(pipe_write_fd) != 0) {
+			sway_log(L_ERROR, "Unable to create pipe for status_command fork (write)");
+			close(pipe_read_fd[0]);
+			close(pipe_read_fd[1]);
```

Comment:
These changes seem unrelated?

Reply:
These are related. 
There were no pipe created for the stdin of a spawned process. The lines marked above are part of a bigger change which defines pipe_write_fd and attaches if to stdin of a status command

Reply:
Ah, fair enough.